src/mixins/filing-template-mixin.ts

```diff
@@ -562,21 +556,11 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     )
 
     // store Name Translations
-    if (filing.incorporationApplication) {
-      // The first time (when we initiate a correction), the `oldName` and `action` props are not
-      // available in the API response, which creates the issue of not having these props in store.
-      // Due to missing props, change event was not triggering if the action value is changed (at
-      // the time of Delete there is no other prop change except action). To handle this scenario,
-      // this structure needs to be kept.
+    if (this.isCorrectedIncorporationApplication) {
       this.setNameTranslations(
-        filing.incorporationApplication.nameTranslations?.map(x => {
-          return {
-            id: x.id,
-            name: x.name,
-            oldName: x.oldName || null,
-            action: x.action || null
-          }
-        }) || []
+        this.mapNameTranslations(filing.correction.nameTranslations) ||
+        this.mapNameTranslations(entitySnapshot.nameTranslations) ||
+        []
```

Comment:
Try to store name translations from draft correction.
Try to store name translations from business snapshot.
Otherwise fall back to empty array.